lib/Sema/CSDiagnostics.cpp

```diff
@@ -8199,6 +8199,44 @@ bool UnableToInferClosureReturnType::diagnoseAsError() {
   return true;
 }
 
+bool UnableToInferGenericPackElementType::diagnoseAsError() {
+  auto *locator = getLocator();
+  auto path = locator->getPath();
+
+  auto packElementElt = (path.end() - 1)->getAs<LocatorPathElt::PackElement>();
```

Comment:
Nit: We don't actually need to do this arithmetic - use `locator->getLastElementAs<LocatorPathElt::PackElement>()` instead.